lib/src/index_impl.dart

```diff
@@ -79,6 +84,19 @@ class MeiliSearchIndexImpl implements MeiliSearchIndex {
     await http.deleteMethod('/indexes/$uid');
   }
 
+  @override
+  Future<MeiliSearchIndex> fetchInfo() async {
+    final index = await client.getIndex(uid);
+    this.primaryKey = index.primaryKey;
```

Comment:
The getINdex method already returns an index instance with the right primary key if I'm correct.

Reply:
Yes, but I thought this function was created in addition to the `Index()` method to allow the update of the primaryKey in the `index` object for example:
```
    var index = index("myIndex");
    index.AddDocuments(mydocument, myPrimaryKey);
    index.fetchInfo();
    print(index.primaryKey);
```
It's working only in this case if I remove the update of the primaryKey:
```
    var index = index("myIndex");
    index.AddDocuments(mydocument, myPrimaryKey);
    var indexUpdate = index.fetchInfo();
    print(indexUpdate.primaryKey);
```
Otherwise, what is the purpose of this function?